lib/pagination.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module Pagination
+  def paginated_result(cursor, page_size, order_column)
+    model = @opts[:model]
+    page_size = (page_size&.to_i || 10).clamp(1, 100)
+    order_column_sym = (order_column || "id").to_sym
+
+    fail Validation::ValidationFailed.new({cursor: "No resource exist with the given id #{cursor}"}) if cursor && !model.from_ubid(cursor)
+    fail Validation::ValidationFailed.new({order_column: "Given order column does not exist for the resource"}) unless model.columns.include?(order_column_sym)
+
+    if cursor
+      cursor_order_column_value = model.from_ubid(cursor).send(order_column_sym)
+      page_records = where(Sequel.qualify(model.table_name, order_column_sym) >= cursor_order_column_value).order(order_column_sym).limit(page_size + 1).all
```

Comment:
I see what you're doing here with page_size + 1. You pop it in line 20 as the next cursor. It was a bit confusing to understand when I look at it. Can you add a comment here?